servers/rendering/rendering_server_default.h

```diff
@@ -681,6 +681,7 @@ class RenderingServerDefault : public RenderingServer {
 
 	FUNC2(viewport_set_use_xr, RID, bool)
 	FUNC3(viewport_set_size, RID, int, int)
+	FUNC1RC(Size2i, viewport_get_size, RID)
```

Comment:
This causes a sync of the rendering thread which is really bad for performance. Which is why this function wasn't exposed in the first place. 

If you need a better way to get the viewport size, just cache the size in the Viewport node at the same time that you call `viewport_set_size()`